app/controllers/deregister_exemptions_controller.rb

```diff
@@ -39,4 +34,11 @@ def handle_permission_denied
     unsuccessful_redirection = WasteExemptionsEngine::ApplicationController::UNSUCCESSFUL_REDIRECTION_CODE
     redirect_to "/pages/permission", status: unsuccessful_redirection
   end
+
+  def process_form_submission(deregistration_service)
+    deregistration_service.deregister!(@deregister_exemptions_form.state_transition)
+    @resource.update_attributes(deregistration_message: @deregister_exemptions_form.message)
```

Comment:
I think traditionally we've done these updates in the form object following a successful `valid?` check, so I'd lean towards doing that again in this case for consistency's sake - is there a reason to do it in the controller instead?

Reply:
Fixed in 1e73e82.